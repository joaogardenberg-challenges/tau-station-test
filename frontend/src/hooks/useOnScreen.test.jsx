import { useRef } from 'react'
import { render, screen } from '@testing-library/react'
import 'intersection-observer'
import useOnScreen from './useOnScreen'

const Component = () => {
  const ref = useRef()
  const isVisible = useOnScreen(ref)

  return <div ref={ref}>{isVisible ? 'visible' : 'not visible'}</div>
}

describe('useOnScreen', () => {
  it('always not visible because it never intersects anything in jsdom', () => {
    const { container } = render(<Component />)
    expect(screen.getByText('not visible')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
