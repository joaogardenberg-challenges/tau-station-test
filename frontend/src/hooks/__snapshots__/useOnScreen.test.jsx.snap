// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOnScreen always not visible because it never intersects anything in jsdom 1`] = `
<div>
  <div>
    not visible
  </div>
</div>
`;
